cpp/source/deal.II/mp_values.h

```diff
@@ -41,107 +41,84 @@ class MPValues
 
   MPValues() = default;
 
-  MPValues(MPValuesParameters<dim, spacedim> const &params);
-
   virtual ~MPValues() = default;
 
   virtual void get_values(std::string const &key,
                           active_cell_iterator const &cell,
-                          std::vector<double> &values) const;
+                          std::vector<double> &values) const = 0;
+};
+
+//////////////////////// NEWER STUFF ///////////////////////////////////////////
+template <int dim, int spacedim = dim>
+class CompositeMat : public MPValues<dim, spacedim>
+{
+public:
+  using active_cell_iterator =
+      typename dealii::DoFHandler<dim, spacedim>::active_cell_iterator;
 
-  virtual void
-  get_values(std::string const &key, active_cell_iterator const &cell,
-             std::vector<dealii::Tensor<1, spacedim>> &values) const;
+  CompositeMat() = default;
+
+  void get_values(std::string const &key, active_cell_iterator const &cell,
+                  std::vector<double> &values) const override;
 
 protected:
   std::unordered_map<dealii::types::material_id, std::shared_ptr<MPValues<dim>>>
-      materials;
+      _materials = {};
```

Comment:
This is C++11 class member initialization. See [here](http://www.stroustrup.com/C++11FAQ.html#member-init).